### PR TITLE
Paris/service/reservation interface tests

### PR DESCRIPTION
**Pull request by PhilippParis**

### CODE REVIEW

File: src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/service/impl/ReservationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import com.at.ac.tuwien.sepm.ss15.edulium.service.ReservationService;
+import com.at.ac.tuwien.sepm.ss15.edulium.service.ServiceException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+/**
+ * implementation of the ReservationService
+ */
+public class ReservationServiceImpl implements ReservationService {
```
Comment (emmanuel099) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/service/impl/ReservationServiceImpl.java]: remove public

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/service/TestReservationService.java [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+        reservationService.addReservation(res);
+
+        // THEN
+        // no single table and no tables next to each other are available
+        // should choose any tables which can host 12 persons
+        int seats = 0;
+        for(Table t : res.getTables()) {
+            seats += t.getSeats();
+        }
+        assertEquals(12, seats);
```
Comment (emmanuel099) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/service/TestReservationService.java]: assertTrue(seats >= res.getQuantity()); ?

e.g. two tables with 8 and 10 seats -> 18 seats but reservation has only 12 seats

File: src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/Main.java [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
+
+        primaryStage.setTitle("FXML Welcome");
+        primaryStage.setScene(scene);
         primaryStage.show();
     }
+
+
+    public static void main(String[] args) {
+        launch(args);
+    }
```
Comment (emmanuel099) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/Main.java]: unrelated changes?!

File: src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/service/ReservationService.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+public interface ReservationService extends Service {
+    /**
+     * returns a list of tables which are free at the time of the reservation
+     * (can be used for manually selecting tables for the reservation)
+     * @param reservation reservation
+     * @return list of tables
+     * @throws ServiceException if an error in the service or persistence layer has occurred
+     * @throws ValidationException if the reservation object is not valid for this action
+     */
+    List<Table> getFreeTables(Reservation reservation) throws ServiceException, ValidationException;
```
Comment (emmanuel099) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/service/ReservationService.java]: IMHO we shouldn't provide such a method, because with this method the user can assign "free" tables to a reservation outside of a transaction, but when a new reservation has been added in the meantime, addReservation will fail because the "free" tables assigned to the reservation may not be free anymore.

Comment (PhilippParis) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/service/ReservationService.java]: but it should be possible for the user to manually select tables for a reservation. And there the system should highlight tables which are free to use.

Comment (emmanuel099) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/service/ReservationService.java]: > possible for the user to manually select tables for a reservation

What is the benefit of it?

Comment (PhilippParis) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/service/ReservationService.java]: Customer: "we would like to sit at the lovley table at the window" ;) 
Or if some tables cannot be moved and some can: the waiter can take that into account
